Tests/MarkdownTests/Interpretive Nodes/AsideTests.swift

```diff
@@ -73,4 +73,134 @@ class AsideTests: XCTestCase {
+            let blockQuote = document.child(at: 0) as! BlockQuote
+            XCTAssertNil(Aside(blockQuote, tagRequirement: .requireAnyLengthTag))
+        }
+
+        do {
+            let source = "> See Also: A different topic."
+            let expectedRootDump = """
+            Document
+            └─ BlockQuote
+               └─ Paragraph
+                  └─ Text "A different topic."
+            """
+            try assertAside(
+                source: source,
+                conversionStrategy: .requireAnyLengthTag,
+                expectedKind: .init(rawValue: "See Also")!,
+                expectedRootDump: expectedRootDump)
+        }
+
+        do {
+            let source = "> Important: This is an aside."
+            let expectedRootDump = """
+            Document
+            └─ BlockQuote
+               └─ Paragraph
+                  └─ Text "This is an aside."
+            """
+            try assertAside(
+                source: source,
+                conversionStrategy: .requireAnyLengthTag,
+                expectedKind: .init(rawValue: "Important")!,
+                expectedRootDump: expectedRootDump)
+        }
+    }
+
+    func testConversionStrategyAllowNoLabel() throws {
+        do {
+            let source = """
+            > This is a regular block quote.
+            """
+            let expectedRootDump = """
+            Document
+            └─ BlockQuote
+               └─ Paragraph
+                  └─ Text "This is a regular block quote."
+            """
+            try assertAside(
+                source: source,
+                conversionStrategy: .tagNotRequired,
+                expectedKind: .note,
+                expectedRootDump: expectedRootDump)
+        }
+
+        do {
+            let source = "> See Also: A different topic."
+            let expectedRootDump = """
+            Document
+            └─ BlockQuote
+               └─ Paragraph
+                  └─ Text "A different topic."
+            """
+            try assertAside(
+                source: source,
+                conversionStrategy: .tagNotRequired,
+                expectedKind: .init(rawValue: "See Also")!,
+                expectedRootDump: expectedRootDump)
+        }
+
+        do {
+            let source = "> Important: This is an aside."
+            let expectedRootDump = """
+            Document
+            └─ BlockQuote
+               └─ Paragraph
+                  └─ Text "This is an aside."
+            """
+            try assertAside(
+                source: source,
+                conversionStrategy: .tagNotRequired,
+                expectedKind: .init(rawValue: "Important")!,
+                expectedRootDump: expectedRootDump)
+        }
+    }
+
+    func assertAside(source: String, conversionStrategy: Aside.TagRequirement, expectedKind: Aside.Kind, expectedRootDump: String, file: StaticString = #file, line: UInt = #line) throws {
```

Comment:
I _think_ the aside parsing is losing the source URL location, if the original markup had any.

If I change this function to
```diff
+ let fakeFileLocation = URL(fileURLWithPath: "/path/to/some-file.md")
  let document = Document(parsing: source, source: fakeFileLocation)
  let blockQuote = document.child(at: 0) as! BlockQuote
  let aside = try XCTUnwrap(Aside(blockQuote, tagRequirement: conversionStrategy), file: file, line: line)
+ XCTAssertEqual(
+     blockQuote.range?.lowerBound.source,
+     aside.content.first?.range?.lowerBound.source,
+     "The parsed aside shouldn't lose its source information",
+     file: file, line: line
+ )
  XCTAssertEqual(expectedKind, aside.kind, file: file, line: line)
- XCTAssertEqual(expectedRootDump, aside.content[0].root.debugDescription(), file: file, line: line)
+ XCTAssertEqual(expectedRootDump, aside.content[0].root.debugDescription(options: .printSourceLocations), file: file, line: line)
```

I get failures for every call site that `"Optional(file:///path/to/some-file.md)") is not equal to ("nil")`

Also, the expected root dumps all pass, despite dumping the aside with `debugDescription(options: .printSourceLocations)`. 

I think together these mean that the aside's markup elements don't have a source location anymore.

The effect of this will be that other tools can't present diagnostics about the content of the parsed aside and point to the location in the source file.